src/main/java/hwr/oop/riddler/logic/backtrackingsolver/component/SinglePossibleImplementor.java

```diff
@@ -0,0 +1,39 @@
+package hwr.oop.riddler.logic.backtrackingsolver.component;
+
+import hwr.oop.riddler.model.Sudoku;
+import hwr.oop.riddler.model.component.Cell;
+
+public class SinglePossibleImplementor implements SolvingComponent {
+    private Sudoku sudoku;
+
+    @Override
+    public boolean execute(Sudoku sudoku) {
+        this.sudoku = sudoku;
+        boolean changesMade = false;
+
+        for (Cell cell : this.sudoku.getAllCells()) {
+            changesMade = fillCellWithOnlyPossibleValue(cell);
+        }
+        return changesMade;
+    }
+
+    private boolean fillCellWithOnlyPossibleValue(Cell cell) {
+        if (cell.isFilled() || !cellHasOnePossibleValue(cell))
+            return false;
+
+        cell.setValue(onlyPossibleValue(cell));
+        return true;
+    }
+
+    private boolean cellHasOnePossibleValue(Cell cell) {
+        return sudoku.getSize() - cell.getImpossibles().size() == 1;
+    }
+
+    private int onlyPossibleValue(Cell cell) {
```

Comment:
Das ist ein duplicate oder?

Reply:
kommt aber später hoffentlich in die Cell Klasse